samcli/lib/utils/async_utils.py

```diff
@@ -113,4 +120,12 @@ def run_async(self):
         List of result of the executions in order
         """
         event_loop = asyncio.new_event_loop()
-        return run_given_tasks_async(self._async_tasks, event_loop)
+        with ThreadPoolExecutor() as self.executor:
+            return run_given_tasks_async(self._async_tasks, event_loop, self.executor)
+
+    def shutdown_and_wait(self):
+        """
+        Will shutdown the not started tasks, and wait the running tasks to be done
+        """
+        if self.executor:
+            self.executor.shutdown()
```

Comment:
We are manually shutting down the executor here, but using it with the context at line 123, will also shutdown it. Do we still need to manually shutdown here?

Reply:
correct, no need for this call. removed

Reply:
Thanks!